arch/eventloop_posix_tcp.c

```diff
@@ -490,6 +490,66 @@ TCP_registerListenSocketDomainName(UA_ConnectionManager *cm, const char *hostnam
     return UA_STATUSCODE_GOOD;
 }
 
+#ifdef _WIN32
+static
+UA_RegisteredFD *
+findRegisteredFD(TCPConnectionManager *cm, uintptr_t connectionId) {
+    UA_RegisteredFD *rfd, *rfd_tmp;
+    LIST_FOREACH_SAFE(rfd, &cm->fds, es_pointers, rfd_tmp) {
+
+        if (rfd->fd == connectionId) {
+            return rfd;
+        }
+    }
+    return NULL;
+}
+
+static
+UA_DelayedCallback *createExplicitlyDelayedShutdownCallback(UA_ConnectionManager *cm, UA_RegisteredFD *rfd, UA_Callback callback) {
+    UA_DelayedCallback *explicitlyDelayedShutdownCallback = (UA_DelayedCallback*) UA_malloc(sizeof(UA_DelayedCallback));
+    if (explicitlyDelayedShutdownCallback == NULL) {
+        return NULL;
+    }
+    explicitlyDelayedShutdownCallback->callback = callback;
+    explicitlyDelayedShutdownCallback->application = cm;
+    explicitlyDelayedShutdownCallback->data = rfd;
+    return explicitlyDelayedShutdownCallback;
+}
+
+static
+void delayedShutdownCallback(void* application, void* data) {
+
+    UA_ConnectionManager *cm = (UA_ConnectionManager*) application;
+    TCPConnectionManager *tcm = (TCPConnectionManager*)cm;
+    UA_RegisteredFD* rfd = (UA_RegisteredFD *) data;
+
+    UA_ByteString response = tcm->rxBuffer;
+    int ret = UA_recv(rfd->fd, (char *)response.data, response.length, MSG_DONTWAIT | MSG_PEEK);
+    if(ret == 0 || ret == -1) {
+        cm->connectionCallback(cm, (uintptr_t)rfd->fd, &rfd->context,
+                               UA_STATUSCODE_BADCONNECTIONCLOSED,
+                               0, NULL, UA_BYTESTRING_NULL);
+        TCP_close(tcm, rfd);
+    } else {
+
+        /* add the delayed shutdown callback again because the peeked data will be processed in the next el cycle */
+        UA_DelayedCallback *explicitlyDelayedShutdownCallback = createExplicitlyDelayedShutdownCallback(cm, rfd, delayedShutdownCallback);
+        if (explicitlyDelayedShutdownCallback == NULL) {
+            UA_LOG_SOCKET_ERRNO_WRAP(
+                UA_LOG_ERROR(cm->eventSource.eventLoop->logger,
```

Comment:
This is rather a warning.
Because we just get some additional delay (120s).
But there is no major disruption.

Also add a comment that if `createExplicitlyDelayedShutdownCallback` fails, the connection will still be closed eventually.